invenio_rdm_records/contrib/meeting/custom_fields.py

```diff
@@ -0,0 +1,119 @@
+#
+# Copyright (C) 2023 CERN.
+#
+# Invenio-RDM-Records is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+"""Meeting custom fields.
+
+Implements the following fields:
+
+- meeting.acronym
+- meeting.dates
+- meeting.place
+- meeting.session_part
+- meeting.session
+- meeting.title
+- meeting.url
+"""
+
+from flask_babelex import _
+from invenio_records_resources.services.custom_fields import BaseCF
+from marshmallow import fields, validate
+from marshmallow_utils.fields import SanitizedUnicode
+
+
+class MeetingCF(BaseCF):
+    """Nested custom field."""
+
+    @property
+    def field(self):
+        """Meeting fields definitions."""
+        return fields.Nested(
+            {
+                "acronym": SanitizedUnicode(),
+                "dates": SanitizedUnicode(),
+                "place": SanitizedUnicode(),
+                "session_part": SanitizedUnicode(),
+                "session": SanitizedUnicode(),
+                "title": SanitizedUnicode(),
+                "url": SanitizedUnicode(
+                    validate=validate.URL(error="You must provide a valid URL."),
+                ),
+            }
+        )
+
+    @property
+    def mapping(self):
+        """Meeting search mappings."""
+        return {
+            "type": "nested",
```

Comment:
is this `nested` or `object` in the end? We might get issues with `nested` (they also have performance implications). So unless is needed for a specific use case (i.e. query patterns) I'd change to `object`.

Reply:
You're right, it should be an `object` since we won't have an array of objects (meetings) in the same document.

Reply:
About the RelationshipCF, Alex mentioned we cannot do with object. I do not recall why (but there is also a comment mentioning this in the code 😅 we just miss the reason. @slint ?

Reply:
This one is `nested` on purpose :)

It's arbitrary key/value list pairs that are represented, which wouldn't work with object.